### PR TITLE
py sol for game routes

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: https://cses.fi/paste/456131b96e64ba3aa3275f/

### CODE REVIEW

File: solutions/gold/cses-1681.mdx [hunk 1]
```diff
@@ -139,4 +144,48 @@ public class cses1681 {
...
 ```
 </JavaSection>
+
+<PySection>
+
+```py
+from collections import deque
+
+edge = [[] for _ in range(100001)]
+backedge = [[] for _ in range(100001)]
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1681.mdx:154]: magic number again

File: solutions/gold/cses-1681.mdx [hunk 2]
```diff
@@ -5,9 +5,9 @@ title: Game Routes
 author: Andrew Wang, Sofia Yang
 ---
 
-**Time Complexity:** $\mathcal{O}(N+M)$
+## Solution
 
-This problem is very similar to the "Longest Flight Route" problem discussed
+This problem is very similar to the "[Longest Flight Route](https://usaco.guide/gold/toposort?lang=py#solution---longest-flight-route)" problem discussed
```
Comment (ryanchou-dev) [inline solutions/gold/cses-1681.mdx:10]:
```suggestion
This problem is very similar to the "[Longest Flight Route](/gold/toposort#solution---longest-flight-route)" problem discussed
```

File: solutions/gold/cses-1681.mdx [hunk 3]
```diff
@@ -139,4 +144,48 @@ public class cses1681 {
...
+while q:
+	node = q.popleft()
+
+	for next_node in edge[node]:
+		in_degree[next_node] -= 1
+		if in_degree[next_node] == 0:
+			q.append(next_node)
+
+	for prev_node in backedge[node]:
+		dp[node] = (dp[node] + dp[prev_node]) % 1000000007
```
Comment (ryanchou-dev) [inline solutions/gold/cses-1681.mdx:185]: here too

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/gold/cses-1681.mdx (hunk 3 above)
Comment (freakin23) [reply solutions/gold/cses-1681.mdx]: done

File: solutions/gold/cses-1681.mdx (hunk 1 above)
Comment (freakin23) [reply solutions/gold/cses-1681.mdx]: done

File: solutions/gold/cses-1681.mdx [hunk 4]
```diff
@@ -139,4 +144,51 @@ public class cses1681 {
...
+<PySection>
+
+```py
+from collections import deque
+
+MAX_N = 10**5 + 1
+MOD = 10**9 + 7
+
+edge = [[] for _ in range(MAX_N)]
+backedge = [[] for _ in range(MAX_N)]
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1681.mdx:157]: why not just initialize in terms of `n` instead

Comment (freakin23) [reply solutions/gold/cses-1681.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**